### PR TITLE
feat: adds a file structure sidebar in the editor

### PR DESCRIPTION
**Pull request by CF3B5**
## Description
The configuration file is too long, so I did this feature, clicking on the structure tree item will jump to the number of file lines.

## Related Tickets & Documents
none.

## Mobile & Desktop Screenshots/Recordings
Voron-2-4
the sidebar is hidden on mobile screen.

## [optional] Are there any post-deployment tasks we need to perform?
none.

Signed-off-by: cf3b5 <cf3b5bb@gmail.com>

### CODE REVIEW

**Comment by meteyou** [comment]
thanks for the PR! i'm looking forward to testing it. if you need any help or information from me about the prettier test, feel free to ping me!

**Comment by CF3B5** [comment]
> thanks for the PR! i'm looking forward to testing it. if you need any help or information from me about the prettier test, feel free to ping me!

I try my best. Why do I always fail the prettier test. :-(

**Comment by meteyou** [comment]
@CF3B5 just try to run `npm run format` and commit & push the changes...

**Comment by meteyou** [comment]
This PR should also fix #1320

**Comment by meteyou** [comment]
Many thanks for this PR! I really love this function!

I just have one small suggestion for improvement. Would it be possible to jump to the section when clicking on the name and only open/close the attributes when clicking on the triangle?

**Comment by CF3B5** [comment]
> Many thanks for this PR! I really love this function!
> 
> I just have one small suggestion for improvement. Would it be possible to jump to the section when clicking on the name and only open/close the attributes when clicking on the triangle?

ok, I've changed. thank you for liking this feature :-)

File: src/components/TheEditor.vue [hunk 1]
```diff
@@ -31,6 +31,16 @@
                         <v-icon small class="mr-1">{{ mdiHelp }}</v-icon>
                         {{ $t('Editor.ConfigReference') }}
                     </v-btn>
+                    <v-btn
+                        v-if="configFileStructure"
+                        text
+                        tile
+                        target="_blank"
```
[the remark this answers is not in the document]
Comment (CF3B5) [reply src/components/TheEditor.vue]: Yes, I forgot to delete it. thank you

File: src/components/TheEditor.vue [hunk 2]
```diff
@@ -49,12 +59,46 @@
...
                 <v-card-text class="pa-0">
-                    <codemirror-async
-                        v-if="show"
-                        ref="editor"
-                        v-model="sourcecode"
-                        :name="filename"
-                        :file-extension="fileExtension" />
+                    <div
+                        v-if="fileStructureSidebar"
+                        class="d-none d-md-flex float-right structure-sidebar cm-editor ͼo">
```
[the remark this answers is not in the document]
Comment (CF3B5) [reply src/components/TheEditor.vue]: I tried it at first,when codemirror width 100% or flex-grow-1, and there was a really weird bug.
[Voron 2.4.webm](https://github.com/user-attachments/assets/198672e1-4928-4b37-ab47-eeac56525720)
So for now I can only use the codemirror width calc(100% - 300px).

File: src/components/TheEditor.vue [hunk 3]
```diff
@@ -398,4 +509,14 @@ export default class TheEditor extends Mixins(BaseMixin) {
     background-color: var(--color-primary);
     background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' viewBox='0 0 24 24'%3E %3Cpath d='M15.88 8.29L10 14.17l-1.88-1.88a.996.996 0 1 0-1.41 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7a.996.996 0 0 0 0-1.41c-.39-.39-1.03-.39-1.42 0z' fill='%23fffff'/%3E %3C/svg%3E");
 }
+
+@media screen and (min-width: 960px) {
+    .structure {
```
[the remark this answers is not in the document]
Comment (CF3B5) [reply src/components/TheEditor.vue]: So, because codemirror width requires calc(100%-300px), this has only been changed a little.

**Review by meteyou — APPROVED** [review]

**Comment by meteyou** [comment]
@CF3B5 thanks for your PR!

i just fixed some visual issues in the treeview and refactored some code blocks.

**Pull request merged**